fastestimator/trace/io/model_saver.py

```diff
@@ -12,10 +12,31 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 # ==============================================================================
+from typing import Union
 
+import tensorflow as tf
+import torch
+
+from fastestimator.backend.save_model import save_model
 from fastestimator.trace.trace import Trace
+from fastestimator.util import Data
 
 
 class ModelSaver(Trace):
-    def __init__(self, model):
-        pass
+    """save model weights based on epoch frequency during training
+
+    Args:
+        model: model instance
+        save_dir: folder path to save model
+        frequency: model saving frequency in epoch(s). Defaults to 1.
+    """
+    def __init__(self, model: Union[tf.keras.Model, torch.nn.Module], save_dir: str, frequency: int = 1):
+        super().__init__(mode="train")
+        self.model = model
+        self.save_dir = save_dir
```

Comment:
should do the save_dir if check as an assert out here rather than every epoch end

Reply:
the purpose of this `save_dir` not for assertion, it is for future integration testing ( when save_dir=None, we won't actually write any model to disk )

Reply:
ah. should probably have an inline comment whenever we're doing something janky for the sake of integration tests so that people don't forget why it's there

Reply:
no need for that in my opinion,  it 's pretty obvious

Reply:
it was not at all obvious to me why we were doing it, esp when the typing for save_dir doesn't indicate that None is even an allowed value

Reply:
done